docs/DATA.md

```diff
@@ -0,0 +1,88 @@
+
+# The Data Layer
+
+Users controlling their own data is a core feature of Habitat. To achieve this, each user's personal node must be capable of serving data in a variety of different formats. Its conceivable that user's will be running applications that simultaneously require relational databases, key-value stores, distributed filesystem access, and more. Habitat refers to these different types of data systems as `datastores`. While these different datastores typically have very different access patterns, Habitat attempts to present a unified data layer that will allow applications to access the datastores through a single API (we need a nice name for this).
+
+```
+Note: This system is probably important enough that we give it its own name, that is not just purely descriptive. 
+```
+
+Why have a single interface for so many different types of datastores? After all, expected data both in and out varies greatly between the different types of datastores. However, there are several major advantages of a single data access layer across all of Habitat:
+1. Interoperable applications are way easier to build if they all use the same API for accessing data.
+2. Consolidating the access point for these datastores reduces the number of ports that have to be opened on each node, simplifying the system.
+3. Common proxying logic can be implemented to forward requests for data not immediately available on the queried node's machine. 
+4. A common authorization and permissions system can be implemented as middleware for this data access API. This also increases the interoperability of applications.
+
+## The API
+
+The data layer is served over `http`. Using the Habitat proxy, HTTP requests can be made over a `LibP2P` transport layer to bypass NATs. The URL for accessing data on any datastore follows this pattern:
+
+`https://<node_host>:<reverse_proxy_port>/data/<community_id>/<datastore>/<namespace>/<content_identifer>`
+
+The components:
+* `node_host`: IP or domain name for the node being queried
+* `reverse_proxy_port`: The node's reverse proxy port, default 2041
+* `community_id`: The id for the community storing the data you are trying to access
```

Comment:
what if this is null? e.g. you are requesting data stored on your own node locally

Reply:
I think this would be simplified a lot if each node automatically comes with a default "node community", that has some higher level controller that prevents members from being added. Similarly, we should have a "user community" that has a similar restriction, but allows multiple nodes that have the same user pk.

Reply:
I think client-side we'll have to do some fanciness to abstract this away - like for requesting "user's data" vs "community data" - we don't want the frontend to have to know about each community id right? or is that okay?